sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/FileFormatWriter.scala

```diff
@@ -111,9 +111,18 @@ object FileFormatWriter extends Logging {
     job.setOutputValueClass(classOf[InternalRow])
     FileOutputFormat.setOutputPath(job, new Path(outputSpec.outputPath))
 
-    val allColumns = queryExecution.logical.output
+    val allColumns = queryExecution.executedPlan.output
```

Comment:
This is problematic.  The physical plan may have different schema from logical plan(schema name may be different), and the writer should respect the logical schema as that what users expects.

Reply:
Yes. We should always use `analyzed.output`